xbmc/guilib/Shader.cpp

```diff
@@ -151,8 +151,13 @@ bool CGLSLVertexShader::Compile()
   {
     GLchar log[LOG_SIZE];
     CLog::Log(LOGERROR, "GL: Error compiling vertex shader");
-    glGetShaderInfoLog(m_vertexShader, LOG_SIZE, NULL, log);
-    CLog::Log(LOGERROR, "{}", log);
+    GLsizei length;
+    glGetShaderInfoLog(m_vertexShader, LOG_SIZE, &length, log);
```

Comment:
apparently it's possible to dynamically allocate the required size of the log by using `GL_INFO_LOG_LENGTH` with `glGetShaderiv`. Might be something worth looking into in the future instead of having a static buffer.